### PR TITLE
Support OTC metadata parsing

### PR DESCRIPTION
**Pull request by karras**
This PR adds support for OTC metadata parsing.

### CODE REVIEW

File: parser_test.go [hunk 1]
```diff
@@ -404,7 +404,8 @@ const exampleStateFile = `
 							"access_ip_v4": "10.0.0.19",
 							"access_ip_v6": "",
 							"tag.%": "1",
-							"tag.tfinventory": "rocks"
+							"tag.tfinventory": "rocks",
+                            "metadata.superman": "clarkkent"
```
Comment (adammck) [inline parser_test.go]: Please fix indentation. This file uses tabs.

File: parser_test.go [hunk 2]
```diff
@@ -548,7 +549,8 @@ const expectedListOutput = `
 	"status_superserver": ["10.120.0.226"],
 	"database": ["10.0.0.8"],
 	"scw_test": ["10.0.0.11"],
-	"tfinventory_rocks": ["10.0.0.19"]
+	"tfinventory_rocks": ["10.0.0.19"],
+    "superman_clarkkent": ["10.0.0.19"]
```
Comment (adammck) [inline parser_test.go]: Please fix indentation. This file uses tabs.

File: parser_test.go (hunk 1 above)
Comment (karras) [reply parser_test.go]: Good catch, thanks! It's fixed now. :)

File: parser_test.go (hunk 2 above)
Comment (karras) [reply parser_test.go]: Good catch, thanks! It's fixed now. :)

**Comment by adammck** [comment]
Thank you!

**Pull request merged**